core/sequencer/runner/native.go

```diff
@@ -118,3 +86,39 @@ func DoMapMapLeafFn(fn MapMapLeafFn, leaves []*tpb.MapLeaf) ([]*entry.IndexedVal
 	}
 	return outs, nil
 }
+
+// ReduceMutationFn takes all the mutations for an index and an auxiliary input
+// of existing mapleaf(s) returns a new map leaf.  ReduceMutationFn must be
+// idempotent, commutative, and associative. i.e. must produce the same output
+// regardless of input order or grouping, and it must be safe to run multiple
+// times.
+type ReduceMutationFn func(msgs []*pb.EntryUpdate, leaves []*pb.EntryUpdate,
```

Comment:
This type is now defined twice. I'm used to duplication being a smell. Is there a reason this isn't?

Reply:
Go encourages interfaces to be declared near the interface consumers, not the producers:
https://blog.chewxy.com/2018/03/18/golang-interfaces/

This keeps packages separate and not tied together. 

I could be persuaded that the client needs a reference to the `mapper`, but that feels like an odd dependency.

Reply:
Nice. I like the ideology. I guess the smell here then is that they both have the same documentation string, and any updates to it may not be propagated to the other place. I will need to spend more time living away from Java to understand how this situation should be handled. For now I trust your wisdom :-)